mucommander-core/src/main/java/com/mucommander/Application.java

```diff
@@ -399,8 +400,11 @@ private void run() {
             com.mucommander.ui.theme.ThemeManager.loadCurrentTheme();
 
             // Starts Bonjour services discovery (only if enabled in prefs)
-            printStartupMessage("Starting Bonjour services discovery...");
-            com.mucommander.bonjour.BonjourDirectory.setActive(MuConfigurations.getPreferences().getVariable(MuPreference.ENABLE_BONJOUR_DISCOVERY, MuPreferences.DEFAULT_ENABLE_BONJOUR_DISCOVERY));
+            boolean bonjourEnabled = MuConfigurations.getPreferences().getVariable(MuPreference.ENABLE_BONJOUR_DISCOVERY, MuPreferences.DEFAULT_ENABLE_BONJOUR_DISCOVERY);
+            if (bonjourEnabled) {
+                printStartupMessage("Starting Bonjour services discovery...");
+            }
+            CompletableFuture.runAsync(() -> com.mucommander.bonjour.BonjourDirectory.setActive(bonjourEnabled));
```

Comment:
how about moving the async part to the BonjourDirectory class (i.e., making setActive become an async operation) - it might be useful for the Preferences dialog as well

Reply:
will take a look